This is the initial recreation of docs using Antora

settermjd opened the pull request:
While it's hardly complete, it's enough to get started. This PR presents the first, most basic [Antora](https://docs.antora.org/antora/latest/) implementation of the documentation, along with the beginnings of the restructure and rewriting of the documentation.

settermjd committed:
This is the initial restructure of docs with Antora

While it's hardly complete, it's enough to get started. This change
presents the first, most basic Antora implementation of the
documentation, along with the beginnings of the restructure and
rewriting of the documentation.

tyrsson commented on docs-site/antora-playbook.yml:
```diff
@@ -0,0 +1,12 @@
+site:
+  title: The laminas-db documentation
```
Should this be phpdb vs laminas-db?

tyrsson reviewed (commented):
Couple of things.

Do we want the bulk of the "docs-site" eg its style sheets etc in this repo or in the doc-site repo that will be published via github pages?

I would suggest moving all of that, images, css etc to that repo, which I will create in just a moment. Then have it pull this into a build just like it will do the other repo's. Or am I crazy? Feel free to tell me I'm crazy.

If I understood Antora correctly then our playbook would live in the docs.php-db.dev repo and be published to https://docs.php-db.dev (via github pages).

The above and the laminas-db -> phpdb changes are the only things I see that might need to be address.

Thanks for the work on this :)

tyrsson commented:
See the relevant messages in discord regarding the github pages site.

settermjd committed:
Extend the readme to cover contributing to the documentation

This is a modest change at the end of the README to provide the
essentials of contributing to the Antora/AsciiDoc-based documentation.

settermjd committed:
Replace old docs with new version

This change removes the old docs directory containing the
Markdown-based documentation, replacing it with an Antora-based version.

settermjd commented:
> Couple of things.
> 
> Do we want the bulk of the "docs-site" eg its style sheets etc in this repo or in the doc-site repo that will be published via github pages?
> 
> I would suggest moving all of that, images, css etc to that repo, which I will create in just a moment. Then have it pull this into a build just like it will do the other repo's. Or am I crazy? Feel free to tell me I'm crazy.
> 
> If I understood Antora correctly then our playbook would live in the docs.php-db.dev repo and be published to https://docs.php-db.dev (via github pages).
> 
> The above and the laminas-db -> phpdb changes are the only things I see that might need to be address.
> 
> Thanks for the work on this :)

As  I see it, each repo needs to be set up as a [content source](https://docs.antora.org/antora/latest/organize-content-files/) with an accompanying _antora-playbook.yml_ file. Then, [docs.php-db.dev](https://docs.php-db.dev/) can contain the _antora.yml_ file which will bring all of the various content sources together into the holistic documentation.

As for the styling and images, that would definitely be a separate repository.

settermjd committed:
Partially refactor the SQL section of the docs

This is a small/partial change in the SQL docs, which starts to
reorganise this section of the docs into a slightly better structure,
and reworks the docs to help them be easier to read, and friendly to
everyone – especially newcomers.

tyrsson commented:
You mean docs.php-db.dev would contain the playbook and each repo would contain a source marker file antora.yml? Also, the PR needs retargeted to the correct branch. Remember, we started over on versioning when the namespaced changed. The branch you branched this from is WAY behind the default branch. I think the issue is that you have not reforked or updated your default branch.
